scripts/prune_ecr_images.js

```diff
@@ -41,6 +41,50 @@ const options = cli.parse({
   releaseName: [true, 'Name of release', 'string']
 })
 
+const K8S_PAGE_LIMIT = 1
+const ECR_PAGE_LIMIT = 10
+
+const getAllPods = async(k8Client, continueValue, labelSelector, pods = []) => {
+  const matchingPods = await k8Client.listNamespacedPod(
+      'default',
+      'false',
+      false,
+      continueValue,
+      undefined,
+      labelSelector,
+      K8S_PAGE_LIMIT
+  )
+  if (matchingPods?.body?.items) pods = pods.concat(matchingPods.body.items)
+  if (matchingPods.body.metadata?._continue) return getAllPods(k8Client, matchingPods.body.metadata._continue, labelSelector, pods)
+  else return pods
+}
+
+const getAllImages = async(ecr, repoName, token, images=[]) => {
+  const input = {
+    repositoryName: repoName,
+    maxResults: ECR_PAGE_LIMIT
+  }
+  if (token) input.nextToken = token
+  const command = new DescribeImagesCommand(input)
+  const response = await ecr.send(command)
+  if (response.imageDetails) images = images.concat(response.imageDetails)
+  if (response.nextToken) return getAllImages(ecr, repoName, response.nextToken, images)
+  else return images
+}
+
+const deleteImages = async(ecr, toBeDeleted) => {
+  const thisDelete = toBeDeleted.length >= 100 ? toBeDeleted.slice(0, 100) : toBeDeleted
+  const deleteCommand = new BatchDeleteImageCommand({
+    imageIds: thisDelete.map((image) => {
+      return { imageDigest: image.imageDigest }
+    }),
+    repositoryName: options.repoName || 'etherealengine'
+  })
+  await ecr.send(deleteCommand)
+  if (toBeDeleted.length >= 100) return deleteImages(ecr, toBeDeleted.slice(100))
```

Comment:
Do we need an `await` before this `deleteImages`?

Reply:
It wasn't strictly necessary, but I changed it anyway.

If you return an un-awaited function call from a function, the code that calls the top-level function will then be awaiting that inner function call. But it is a bit clearer to await the inner function call as well.